src/nvim/ex_getln.c

```diff
@@ -489,9 +492,52 @@ static void may_do_incsearch_highlighting(int firstc, long count,
     ccline.cmdbuff[skiplen + patlen] = NUL;
     memset(&sia, 0, sizeof(sia));
     sia.sa_tm = &tm;
-    found = do_search(NULL, firstc == ':' ? '/' : firstc,
-                      ccline.cmdbuff + skiplen, count,
-                      search_flags, &sia);
+   
+    // in the case of substitute searches, '/' does not necessarily act as
+    // a delimiter, copy the search pattern and add a backslash to fool do_search
```

Comment:
instead of fooling  `do_search` can't we just tell it?

Reply:
Pardon?

do_search seems to always use '/' as a delimiter, adding the backslashes is in a way just telling `do_search` not to.